### PR TITLE
Add possibility to choose desired arrival time

### PR DESCRIPTION
**Pull request by Elhebert**
Hi,

I would like to be able to know when I have to leave to be home around a given time. With the implementation of the api-key done in 86d4687 and 031622b, you can use transit mode, and choose an `arrival_time` instead of the `departure_time`.

The downside is, it's only available in transit mode, the `arrival_time` paramater is not supported in other mode.

Use: 

```
User> hubot i wanna be home by 16h00
Hubot> From X to Y, you'll have to leave at 15h30
```

### CODE REVIEW

File: src/traffic-status.coffee [hunk 1]
```diff
@@ -48,3 +48,32 @@ module.exports = (robot) ->
 					return
 		else
 			msg.send "You need to set your home and work address in my brain, #{msg.message.user.name}"
+		robot.respond /i wanna be home by (\d\d)(?:h|:)(\d\d)/i, (msg) ->
+		@userbrain = robot.brain.userForName(msg.message.user.name)
+		if not @userbrain.mode
+			@userbrain.mode = "transit"
```
Comment (Cinetik) [inline src/traffic-status.coffee]: Why transit by default while it is driving on the other function ?

File: src/traffic-status.coffee [hunk 2]
```diff
@@ -48,3 +48,33 @@ module.exports = (robot) ->
 					return
 		else
 			msg.send "You need to set your home and work address in my brain, #{msg.message.user.name}"
+	robot.respond /i wanna be home by (\d\d)(?:h|:)(\d\d)/i, (msg) ->
+		@userbrain = robot.brain.userForName(msg.message.user.name)
+		if not @userbrain.mode
+			@userbrain.mode = "transit"
```
Comment (Cinetik) [inline src/traffic-status.coffee]: Why transit by default while it is driving on the other function ?

**Comment by Elhebert** [comment]
My bad, misread a part of the documentation. I'll fix it.

**Comment by Elhebert** [comment]
I corrected the default mode error.

I also added the possibility to set the arrival time in am/pm or in 24h format. Hubot will also now answer the user with the time based on the local set on the server.
An other option that we could implement is to answer with the format the user made his request. If the `arrival_time` is set with am/pm, hubot answer in amp/pm, in 24h format otherwise.

**Pull request merged**